Created the POST endpoint for courses

abdulqshabbir opened the pull request:
API now accepts POST requests on `/courses` and will save a course to the database.

abdulqshabbir committed:
Created the POST endpoint for courses

abdulqshabbir committed:
Created the POST endpoint for courses

amirfakhrullah commented on server/src/controllers/courses/createCourse.ts:
```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from "express";
+
+export default async function createCourse(req: Request, res: Response) {
```
- When creating a course, we should store the `_id` of the course in the `user.courses` arrays
- It should look something like this:
```typescript
export default async function createCourse(req: Request, res: Response) {
   const id = getUserId(req); // helper to extract user id from the session
   const user = await User.findOne({ _id: id });
   
   // some error handling if no user found in db

    try {
        const newCourse = await createCourseInDB(req.body)
        user.courses.push(newCourse._id);
        await user.save();
        return res.status(200).json({ // also make sure the response to be consistent
             success: true,
             data: newCourse
         });
    } catch(e) {
        return res.status(400).json({
            success: false,
            message: e.message
        })
    }
}
```

you can refactor and separate them into services if you want, as long as the behavior is the same as above then it should be fine

AllMikeNoIke replied:
it should be a 201 status code for successfully created content, as well!

abdulqshabbir replied:
Thanks! good call :)

amirfakhrullah commented:
overall looks good :)

AllMikeNoIke commented on server/src/controllers/courses/createCourse.ts:
```diff
@@ -0,0 +1,32 @@
+            end_date: body.end_date,
+            color: body.color,
+            weekly_schedule: body.weekly_schedule
+        })
```
i'd use object destructuring in the parameters here, just to make the code a little cleaner

abdulqshabbir replied:
Sounds good will make the changes!

amirfakhrullah committed:
hotfix: passport

abdulqshabbir committed:
Made small fix to serializeUser
Passed down _id to callback function of serializeUser instead of {_id }.
Also, added some error handling to the deserializeUser function.

abdulqshabbir committed:
Merge pull request #59 from chingu-voyages/hotfix/passport

hotfix: passport

abdulqshabbir committed:
refactored /courses POST route
Separated database logic into a service function.
Changed status code from 200 to 201 for success.
Created consistent return format for data.

abdulqshabbir committed:
added swagger docs for POST route

abdulqshabbir committed:
resolved merge conflict
